src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <Windows.h>
+
+struct InvalidHandleTraits
+{
+    using HandleType = HANDLE;
+    static const HANDLE DefaultHandle;
+    static void Close(HANDLE handle) { CloseHandle(handle); }
+};
+
+// Workaround for VC++ bug https://developercommunity.visualstudio.com/content/problem/33928/constexpr-failing-on-nullptr-v141-compiler-regress.html
+const HANDLE InvalidHandleTraits::DefaultHandle = INVALID_HANDLE_VALUE;
```

Comment:
I understand your motivation for wrapping a handle. However, the performance cost is ignorable.

Reply:
Clarify? Is the perf difference noticeable?

Reply:
Sorry,  typo.  Updated the comments

Reply:
Abstraction is completely free, it generates same code as without abstraction.
See https://godbolt.org/g/y8XzDe
Test.A and B methods.

Reply:
I meant the constructor and destructor unless compiler put them inline

Reply:
They are both inlined.